src/peft/tuners/lora/config.py

```diff
@@ -466,6 +572,10 @@ def __post_init__(self):
             )
             warnings.warn(msg)
 
+        # convert corda_config to dict
+        if self.corda_config and not isinstance(self.corda_config, dict):
```

Comment:
What happens if users pass `init_lora_weights="corda"` but don't pass `corda_config`? Can we use the default `CordaConfig`?

And what if users pass the `corda_config` but don't set `init_lora_weights="corda"`? Probably an error should be raised on that case.